File: week5/Count-Primes/woo.java
package week5.CountPrimes;

/**
 * created by victory_woo on 2020/10/08
 */
public class woo {
    public static void main(String[] args) {
        System.out.println(countPrimes(10));
        System.out.println(countPrimes(0));
        System.out.println(countPrimes(1));
        System.out.println(countPrimes(2));
        System.out.println(countPrimes(3));
    }

    // n보다 작아야 하는데, n보다 작거나 같은 조건으로 찾고 있었음..
    public static int countPrimes(int n) {
        if (n == 0 || n == 1) return 0;
        boolean[] check = new boolean[n + 1];
        for (int i = 2; i <= n; i++) {
            if (check[i]) continue;

            for (int j = i + i; j <= n; j += i) check[j] = true;
        }

        int count = 0;
        for (int i = 2; i < n; i++) {
            if (!check[i]) count++;
        }

        return count;
    }
}

File: week1/Rotate_Array/Woo.kt
package week1.Rotate_Array

/**
 * created by victory_woo on 2020/09/05
 * */
class Woo {
    fun main(args: Array<String>) {
        //rotate(new int[]{1, 2, 3, 4, 5, 6, 7}, 3);
        rotate(intArrayOf(-1, -100, 3, 99), 2)
    }

    private fun rotate(nums: IntArray, k: Int) {
        var count = 0
        var start = 0
        while (count < nums.size) {
            var current = start
            var cache = nums[current]

            do {
                val next = (current + k) % nums.size
                val temp = nums[next]
                nums[next] = cache
                cache = temp
                current = next
                count++
            } while (start != current)
            start++
        }
        for (i in nums) println(i)
    }
}